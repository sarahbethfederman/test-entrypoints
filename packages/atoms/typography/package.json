{
  "name": "@auscred/typography",
  "version": "0.0.1",
  "main": "dist/Typography.js",
  "types": "dist/Typography.d.ts",
  "files": [
    "dist"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@types/react": "^16.4.0",
    "react": "^16.4.0",
    "styled-components": "^3.3.3"
  },
  "scripts": {
    "build": "tsc --pretty"
  }
}
