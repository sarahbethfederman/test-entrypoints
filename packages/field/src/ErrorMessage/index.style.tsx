import styled from 'styled-components';
import { pt } from '@lendi-ui/spacing';

export const Wrapper = styled.div`
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: base-line;
  ${pt('xxxs')};
`;
