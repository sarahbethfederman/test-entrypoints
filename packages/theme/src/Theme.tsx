import * as React from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
// tslint:disable-next-line no-unused-expression
injectGlobal`
  html, body {
    margin: 0;
    padding: 0;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors.shade[700]};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>{children}</Root>
    </ThemeProvider>
  );
}
