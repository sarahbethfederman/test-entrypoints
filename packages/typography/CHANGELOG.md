## [4.1.0] 2020-06-24 [Release Notes]
- Added 'xl' as a size for Body

## [4.0.1] 2019-09-26 [Release Notes](https://creditandfinance.atlassian.net/wiki/spaces/HUB/pages/803930391/Upcoming+Major+Changes)
- Bumped lendi-ui dependencies to latest versions

## [4.0.0] 2019-9-23 2019-8-21 [Release notes](https://creditandfinance.atlassian.net/wiki/spaces/HUB/pages/803930391/Upcoming+Major+Changes)
### Changed
- Uses Theme v7
- Update typescript to version 3.5.2 and react to version 16.8.4 [HUB-151](https://creditandfinance.atlassian.net/browse/HUB-151)

## [3.4.0] - 2019-07-29 - [HUB-338](https://creditandfinance.atlassian.net/browse/HUB-338)
### Added
- Add 'xxs' type of Body, Heading, Link and Overline.

## [3.3.0] - 2019-07-16 - [HUB-330](https://creditandfinance.atlassian.net/browse/HUB-330)
### Changed
- Change link default color to 'currentColor'

## [3.2.1] - 2019-06-4 - [HUB-262](https://creditandfinance.atlassian.net/browse/HUB-262)
### Changed
- Updated utils dependency to get correct types

## [3.2.0] - 2019-05-27 - [HUB-169](https://creditandfinance.atlassian.net/browse/HUB-169)
### Changed
- Support for native props like Aria attributes, Standard HTML Attributes like title, classname, id, role, itemProp, itemID, itemRef

## [3.1.1] - 2019-05-23 - [HUB-201](https://creditandfinance.atlassian.net/browse/HUB-201)
### Changed
- Updated links that are set as a button to inherit background-color from parent.

## [3.1.0] - 2019-05-17 - [HUB-174](https://creditandfinance.atlassian.net/browse/HUB-174)
### Changed
- Module format reverted back to CJS.
### Deleted
- Dropped "module" property from package.json.

## [3.0.2] - 2019-05-09 - [HUB-92](https://creditandfinance.atlassian.net/browse/HUB-92)
### Changed
- Updated CHANGELOG to match required parsing format.

## [3.0.1] - 2019-05-08 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)
### Changed
- Change module main entry point to esm in package.json

## [3.0.0] - 2019-04-30 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)
### Deleted
- Remove the CJS bundles.

## [2.7.1] - 2019-04-10 - [HUB-50](https://creditandfinance.atlassian.net/browse/HUB-50)
### Changed
- Fixed the margin bug for Overline and added some tests for margin alignment in all typography.

## [2.7.0] - 2019-04-10 - [HUB-90](https://creditandfinance.atlassian.net/browse/HUB-90)
### Added
- Added "xl" size variant.

## [2.6.0] - 2019-03-14 - [BRAVO-571](https://creditandfinance.atlassian.net/browse/BRAVO-571)
### Added
- Added `Overline` sub-package, which includes `overline` mixin and `Overline` component.

## [2.5.5] - 2019-03-07 - [BRAVO-652](https://creditandfinance.atlassian.net/browse/BRAVO-652)
### Changed
- fixed bug that caused the error: 'children' doesn't exist on type 'IntrinsicAttributes & LinkProps'
