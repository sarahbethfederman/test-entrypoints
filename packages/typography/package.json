{
  "name": "@lendi-ui/typography",
  "version": "6.0.0-beta.0",
  "source": "src/index.ts",
  "main": "dist/typography.cjs.js",
  "module": "dist/typography.esm.js",
  "files": [
    "dist"
  ],
  "license": "ISC",
  "peerDependencies": {
    "@lendi-ui/theme": "^10.0.0-beta.0",
    "@types/react": "^16.8",
    "@types/styled-components": "^4.1.18",
    "react": "^16.8",
    "styled-components": "^4"
  },
  "dependencies": {
    "@lendi-ui/breakpoint": "^6.0.0-beta.0",
    "@lendi-ui/color": "^6.0.0-beta.0",
    "@lendi-ui/spacing": "^7.1.0-beta.0",
    "@lendi-ui/utils": "^6.0.0-beta.0"
  },
  "devDependencies": {
    "@lendi-ui/theme": "10.0.0-beta.0",
    "@types/react": "^16.4.8",
    "@types/styled-components": "^4.1.18",
    "react": "^16.9.0",
    "styled-components": "^4"
  }
}
