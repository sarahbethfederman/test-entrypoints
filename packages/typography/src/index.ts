import styled, { css } from 'styled-components';

import { select } from '@lendi-ui/theme';
import { gte, map, BreakpointValue, BreakpointValueMap } from '@lendi-ui/breakpoint';
import { fg } from '@lendi-ui/color';

export type Alignment = 'left' | 'center' | 'right' | 'justify';
export type AlignmentOrAlignmentMap = BreakpointValue<Alignment> | BreakpointValueMap<Alignment>;
export type HeadingSize = 'xs' | 'sm' | 'md' | 'lg' | 'xl';
export type BodySize = 'xs' | 'sm' | 'md' | 'lg';
export type LinkSize = 'sm' | 'md' | 'lg';

const alignMixin = (alignment: AlignmentOrAlignmentMap) => {
  return map(alignment, (val) => {
    if (val) {
      return `text-align: ${val};`;
    } else {
      return '';
    }
  });
};

const headingSizeMixin = (size: HeadingSize) => {
  switch (size) {
    case 'xs':
      return css`
        font-size: 15px;
        font-weight: 500;
        line-height: calc(20 / 15);

        ${gte('tablet')`
          font-size: 18px;
          line-height: calc(24/18);
        `};
      `;
    case 'sm':
      return css`
        font-size: 18px;
        font-weight: 500;
        line-height: calc(24 / 18);

        ${gte('tablet')`
          font-size: 22px;
          line-height: calc(28/22);
        `};
      `;
    case 'md':
      return css`
        font-size: 22px;
        font-weight: 500;
        line-height: calc(28 / 22);

        ${gte('tablet')`
          font-size: 28px;
          line-height: calc(32/28);
        `};
      `;
    case 'lg':
      return css`
        font-size: 28px;
        font-weight: 500;
        line-height: calc(34 / 28);

        ${gte('tablet')`
          font-size: 37px;
          line-height: calc(40/37);
        `};
      `;
    case 'xl':
      return css`
        font-size: 37px;
        font-weight: bold;
        line-height: calc(40 / 37);

        ${gte('tablet')`
          font-size: 46px;
          line-height: calc(50/46);
        `};
      `;
  }
};

const bodySizeMixin = (size: BodySize) => {
  switch (size) {
    case 'xs':
      return css`
        font-size: 10px;
        line-height: calc(16 / 10);

        ${gte('tablet')`
          font-size: 13px;
          line-height: calc(20/13);
        `};
      `;
    case 'sm':
      return css`
        font-size: 12px;
        line-height: calc(16 / 12);

        ${gte('tablet')`
          font-size: 14px;
          line-height: calc(20/14);
        `};
      `;
    case 'md':
      return css`
        font-size: 14px;
        line-height: calc(20 / 14);

        ${gte('tablet')`
          font-size: 16px;
          line-height: calc(24/16);
        `};
      `;
    case 'lg':
      return css`
        font-size: 16px;
        line-height: calc(24 / 16);

        ${gte('tablet')`
          font-size: 18px;
          line-height: calc(24/18);
        `};
      `;
  }
};

export interface HeadingOptions {
  color?: string;
  align?: Alignment;
  size: HeadingSize;
}

export function heading(options: HeadingOptions) {
  const { color = 'shade.700', align, size } = options;
  return css`
    ${color && fg(color)}
    ${align && alignMixin(align)}
    ${headingSizeMixin(size)}
    font-family: ${select('typography.heading.fontFamily', '"Cabin", sans-serif')};
  `;
}

export interface BodyOptions {
  color?: string;
  align?: Alignment;
  size?: BodySize;
}

/**
 * Body mixin
 */
export function body(options: BodyOptions = {}) {
  const { color = 'shade.700', align, size = 'md' } = options;
  return css`
    ${color && fg(color)}
    ${align && alignMixin(align)}
    ${size && bodySizeMixin(size)}
    font-family: ${select('typography.body.fontFamily', '"Open sans", sans-serif')};
  `;
}

/**
 * Body component
 */
export const Body = styled.p`
  ${link};
`;

export interface LinkOptions {
  color?: string;
  href?: string;
  size?: LinkSize;
}

/**
 * Link mixin
 */
export function link(options: LinkOptions = {}) {
  const { color = 'primary.500', size } = options;

  return css`
    ${color && fg(color)}
    ${size && bodySizeMixin(size)}
    font-family: ${select('typography.body.fontFamily', '"Open sans", sans-serif')};
    font-weight: bold;
    cursor: pointer;

    &:hover,
    &:focus,
    &:active {
      text-decoration: underline;
      ${fg('primary.700')}
    }
  `;
}

/**
 * Link component
 */
export const Link = styled.a`
  ${link};
`;
