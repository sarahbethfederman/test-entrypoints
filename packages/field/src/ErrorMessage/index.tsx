import * as React from 'react';
import { Wrapper } from './index.style';
import { HighlightOff } from '@lendi-ui/icon';
import { Body } from '@lendi-ui/typography';

export interface ErrorMessageProps {
  error?: string;
}

const ErrorMessage = ({ error }: ErrorMessageProps) => {
  const ErrorMessageColor = 'error.500';
  return (
    <Wrapper color={ErrorMessageColor}>
      <HighlightOff color={ErrorMessageColor} />
      <Body size="xs" color={ErrorMessageColor} ml="xxxs">
        {error}
      </Body>
    </Wrapper>
  );
};

export default ErrorMessage;
