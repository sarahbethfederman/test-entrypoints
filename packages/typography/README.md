# @lendi-ui/typography

Typography including Body, Heading and Link.

## Installation

```
yarn add @lendi-ui/typography
```

## Usage

- Body

```jsx
import { Body } from '@lendi-ui/typography';
<Body size="lg" color="warn.500">
  The quick brown fox jumps over the lazy dog
</Body>;
```

- Heading

```jsx
import { Heading } from '@lendi-ui/typography';
<Heading size="lg" color="warn.500">
  The quick brown fox jumps over the lazy dog
</Heading>;
```

- Link

```jsx
import { Link } from '@lendi-ui/typography';
<Link>quick brown fox</Link>;
```
