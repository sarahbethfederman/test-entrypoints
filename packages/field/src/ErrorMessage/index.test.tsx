import * as React from 'react';
import { mount } from 'enzyme';
import ErrorMessage from './index';
import { Body } from '@lendi-ui/typography';
import { Wrapper } from './index.style';
import Theme from '@lendi-ui/theme';

describe('ErrorMessage', () => {
  let wrapper: any;
  beforeEach(() => {
    wrapper = mount(
      <Theme>
        <ErrorMessage error="error message" />
      </Theme>
    );
  });

  it('should render whole component', () => {
    expect(wrapper.find('ErrorMessage').length).toEqual(1);
    expect(wrapper.find('ErrorMessage')).toMatchSnapshot();
  });

  it('should render a div called "Wrapper"', () => {
    expect(wrapper.find(Wrapper).length).toEqual(1);
    expect(wrapper.find(Wrapper)).toMatchSnapshot();
  });

  it('should render a icon called "HighlightOff"', () => {
    expect(wrapper.find('SvgHighlightOff').length).toEqual(1);
    expect(wrapper.find('SvgHighlightOff')).toMatchSnapshot();
  });

  it('should render a paragraph use LUI Body called "MessageBody"', () => {
    expect(wrapper.find(Body).length).toEqual(1);
    expect(wrapper.find(Body).prop('children')).toEqual('error message');
    expect(wrapper.find(Body)).toMatchSnapshot();
  });
});
