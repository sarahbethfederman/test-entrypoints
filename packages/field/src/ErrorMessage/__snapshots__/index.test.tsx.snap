// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage it should render a div called "Wrapper" 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: base-line;
  -webkit-box-align: base-line;
  -ms-flex-align: base-line;
  align-items: base-line;
  padding-top: calc(0.25 * var(--lendi-ui-size));
}

.c1 {
  width: 1em;
  height: 1em;
  vertical-align: middle;
}

.c1 > svg {
  fill: #ff1c4a;
  width: inherit;
  height: inherit;
}

.c2 {
  margin-top: 0;
  margin-bottom: 0;
  color: #ff1c4a;
  font-size: calc(0.75 * var(--lendi-ui-size));
  line-height: calc(16 / 12);
  font-family: 'Open Sans',sans-serif;
  margin-left: calc(0.25 * var(--lendi-ui-size));
}

<styled.div
  color="error.500"
>
  <div
    className="c0"
    color="error.500"
  >
    <HighlightOff
      color="error.500"
    >
      <styled.span
        color="error.500"
      >
        <span
          className="c1"
          color="error.500"
        >
          <svg
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
        </span>
      </styled.span>
    </HighlightOff>
    <styled.p
      color="error.500"
      ml="xxxs"
      size="xs"
    >
      <p
        className="c2"
        color="error.500"
        size="xs"
      >
        error message
      </p>
    </styled.p>
  </div>
</styled.div>
`;

exports[`ErrorMessage it should render a icon called "HighlightOff" 1`] = `
.c0 {
  width: 1em;
  height: 1em;
  vertical-align: middle;
}

.c0 > svg {
  fill: #ff1c4a;
  width: inherit;
  height: inherit;
}

<HighlightOff
  color="error.500"
>
  <styled.span
    color="error.500"
  >
    <span
      className="c0"
      color="error.500"
    >
      <svg
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
    </span>
  </styled.span>
</HighlightOff>
`;

exports[`ErrorMessage it should render a paragraph use LUI Body called "MessageBody" 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  color: #ff1c4a;
  font-size: calc(0.75 * var(--lendi-ui-size));
  line-height: calc(16 / 12);
  font-family: 'Open Sans',sans-serif;
  margin-left: calc(0.25 * var(--lendi-ui-size));
}

<styled.p
  color="error.500"
  ml="xxxs"
  size="xs"
>
  <p
    className="c0"
    color="error.500"
    size="xs"
  >
    error message
  </p>
</styled.p>
`;

exports[`ErrorMessage it should render whole component 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: base-line;
  -webkit-box-align: base-line;
  -ms-flex-align: base-line;
  align-items: base-line;
  padding-top: calc(0.25 * var(--lendi-ui-size));
}

.c1 {
  width: 1em;
  height: 1em;
  vertical-align: middle;
}

.c1 > svg {
  fill: #ff1c4a;
  width: inherit;
  height: inherit;
}

.c2 {
  margin-top: 0;
  margin-bottom: 0;
  color: #ff1c4a;
  font-size: calc(0.75 * var(--lendi-ui-size));
  line-height: calc(16 / 12);
  font-family: 'Open Sans',sans-serif;
  margin-left: calc(0.25 * var(--lendi-ui-size));
}

<ErrorMessage
  error="error message"
>
  <styled.div
    color="error.500"
  >
    <div
      className="c0"
      color="error.500"
    >
      <HighlightOff
        color="error.500"
      >
        <styled.span
          color="error.500"
        >
          <span
            className="c1"
            color="error.500"
          >
            <svg
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </svg>
          </span>
        </styled.span>
      </HighlightOff>
      <styled.p
        color="error.500"
        ml="xxxs"
        size="xs"
      >
        <p
          className="c2"
          color="error.500"
          size="xs"
        >
          error message
        </p>
      </styled.p>
    </div>
  </styled.div>
</ErrorMessage>
`;
